src/parser/RdfParser.cpp

```diff
@@ -1160,19 +1160,26 @@ void RdfParallelParser<T>::initialize(const string& filename,
       bufferSize.getBytes(), "\\.[\\t ]*([\\r\\n]+)");
   ParallelBuffer::BufferType remainingBatchFromInitialization;
   fileBuffer_->open(filename);
-  if (auto batch = fileBuffer_->getNextBlock(); !batch) {
-    LOG(WARN) << "Empty input to the TURTLE parser, is this what you intended?"
-              << std::endl;
-  } else {
-    RdfStringParser<T> declarationParser{};
-    declarationParser.setInputStream(std::move(batch.value()));
-    while (declarationParser.parseDirectiveManually()) {
+  RdfStringParser<T> declarationParser{};
+  while (true) {
+    if (auto batch = fileBuffer_->getNextBlock()) {
+      declarationParser.setInputStream(std::move(batch.value()));
+      while (declarationParser.parseDirectiveManually()) {
+      }
+      auto remainder = declarationParser.getUnparsedRemainder();
+      if (remainder.empty()) {
+        continue;
+      }
+      this->prefixMap_ = std::move(declarationParser.getPrefixMap());
+      remainingBatchFromInitialization.reserve(remainder.size());
+      ql::ranges::copy(remainder,
+                       std::back_inserter(remainingBatchFromInitialization));
+    } else {
+      AD_LOG_WARN
+          << "Empty input to the TURTLE parser, is this what you intended?"
+          << std::endl;
```

Comment:
What about 
```
while (remainingBatchFromInitialization.empty()) {
  // if ! next Block then warn and break;
  // parse directives and copy remainder to remaining batch
}
// extract prefix map from declaration parser.

This has only 1 break (only because we want the logging), and zero continues.